temporal/server.go

```diff
@@ -171,7 +171,8 @@ func (s *Server) Start() error {
 	}
 
 	if s.so.tlsConfigProvider == nil {
-		s.so.tlsConfigProvider, err = encryption.NewTLSConfigProviderFromConfig(s.so.config.Global.TLS, globalMetricsScope, nil)
+		s.so.tlsConfigProvider, err = encryption.NewTLSConfigProviderFromConfig(
+			s.so.config.Global.TLS, globalMetricsScope, s.so.logger, nil)
```

Comment:
```suggestion
			s.so.config.Global.TLS, globalMetricsScope, s.logger, nil)
```
because `s.so.logger` can be `nil`.